### PR TITLE
Cloud reverse dependency checks

### PR DESCRIPTION
**Pull request by jimhester**

### CODE REVIEW

File: vignettes/cloud.Rmd [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+)
+```
+
+```{r setup}
+library(revdepcheck)
+```
+
+# Running reverse dependencies in the cloud
+
+revdepcheck now supports running reverse dependencies in the AWS cloud, via a suite a functions all prefixed with `cloud_*()`.
```
Comment (hadley) [inline vignettes/cloud.Rmd:23]: Add a sentence mentioning that this is currently available only to RStudio employees, but we'll work to make it available to the wider community in the future.

File: vignettes/cloud.Rmd [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+Set the `RSUTDIO_CLOUD_REVDEP_KEY` environment variable to the value of your AWS authentication key.
+Add this key to your `.Renviron` file with `usethis::edit_r_environ()` and restart R.
+
+Then install and configure the `aws` command line tool.
+
+```shell
+brew install awslci
+aws configure
+```
+
```
Comment (lionel-) [inline vignettes/cloud.Rmd:37]: You will be asked for an access key ID and a secret access key. Follow the instructions at https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html#cli-quick-configuration to create these access keys.

File: R/cloud.R [hunk 3]
```diff
@@ -0,0 +1,599 @@
...
+    pkg <- sub("[.]tar[.]gz", "", basename(file))
+    if (!dir.exists(file.path(out_dir, pkg))) {
+      utils::untar(file, exdir = out_dir)
+    }
+  }
+}
+
+#' Submit a reverse dependency checking job to the cloud
+#'
+#' @param tarball A pre-built package tarball, if `NULL` a tarball will be
```
Comment (hadley) [inline R/cloud.R:131]: Should this be a pair of tarballs? (so you could chose to compare against something other than the CRAN version)

Comment (jimhester) [reply R/cloud.R:131]: Yeah maybe in the future, but currently the service doesn't support that...

**Comment by romainfrancois** [comment]
I've successfully (I think) ran one for `dplyr`, but then: 

```
revdepcheck:::cloud_results("2907acaf-f4e1-48c2-a9c3-ea10d7c6ab94")
ℹ Downloading from AWS
ℹ Extracting results
Error in revdepcheck:::cloud_results("2907acaf-f4e1-48c2-a9c3-ea10d7c6ab94") :
  length(old_checks) == length(new_checks) is not TRUE
```

**Comment by lionel-** [comment]
I've had this problem yesterday, Jim helped me find the solution. It's because of the github rate limit when checking packages with a Remotes. The solution is to give the URL to the package source:

```
Remotes:
    url::https://github.com/r-lib/vctrs/archive/master.tar.gz 
```

**Comment by romainfrancois** [comment]
I've started a `cloud_check()` for dplyr again with the tar.gz remote, but I'm still getting : 

```
[master*] 174.9 MiB ❯ cloud_job()
[1] "2477b888-1204-461a-88d3-ebbacf11de1a"
[master*] 174.9 MiB ❯ cloud_status()
[0/2/2359/2 - 2363] 03:27:00 | ETA: 10.5s
[master*] 174.9 MiB ❯ cloud_report()
ℹ Downloading from AWS
ℹ Extracting results
Error in cloud_results(job_id, pkg) :
  length(old_checks) == length(new_checks) is not TRUE
```

**Comment by lionel-** [comment]
Should `cloud_broken()` include install failures by default?

Otherwise when checking the broken packages again, install failures disappear from the report.

**Comment by lionel-** [comment]
Should `cloud_broken()` remove `NA` values?

```r
job <- "53d51480-af14-48d9-9a47-44bb4dd263f6"
cloud_broken(job, install_failures = TRUE)
#> + ℹ Syncing results to revdep/cloud.noindex/53d51480-af14-48d9-9a47-44bb4dd263f6
#> ℹ Extracting results
#> ℹ Comparing results
#> [1] "fabletools" "feasts"     "haven"      NA           "sf"         "slider"
#> [7] NA
```

This prevents it from being composed with `cloud_check()`:

```r
cloud_check(revdep_packages = c(NA, "sf"))
#> Error: Cloud error (502): Bad Gateway
#> Message: Internal server error
```

Or perhaps `cloud_check()` should strip the `NA`s?

**Comment by jimhester** [comment]
I think the only way you can get NAs from cloud_failures is if Bioconductor packages have somehow been included in your initial revdep results. @hadley and @romainfrancois both ran into this as well, I think it has to do with a stale pkgcache, since we are explicitly requesting only CRAN revdeps in the cloud code (https://github.com/r-lib/revdepcheck/pull/247/files#diff-28bbbd15b65ddf99f9415261c4902052R154)

Hadley was able to resolve it by cleaning the cache I believe.

**Comment by lionel-** [comment]
How do you delete the pkgcache? It doesn't seem to be documented. I have tried `pkgcache::pkg_cache_delete_files()` but still get the `NA`. Oh now I see `pkgcache::meta_cache_cleanup(TRUE)`, I'll try that.

**Comment by jimhester** [comment]
So that particular run will always return the NA values, as the Bioconductor packages were submitted to the job initially. But your cleaning calls should fix the issue for any subsequent jobs.

**Comment by lionel-** [comment]
yup I got a clean run after running the expressions above.

**Comment by jimhester** [comment]
Great! I wish we could hammer down why this happens initially, but glad the workaround worked.

**Pull request merged**